fivox/spikeLoader.cpp

```diff
@@ -60,62 +59,40 @@ class SpikeLoader::Impl
         }
 
         const std::string& spikePath = params.getSpikes();
-        _loadSpikes( spikePath.empty() ? params.getConfig().getSpikeSource()
-                                       : URI( spikePath ));
-
+        _report.reset(
+            new brain::SpikeReportReader( spikePath.empty()
+                                          ? params.getConfig().getSpikeSource()
+                                          : URI( spikePath ),
+                                          gids ));
+        _spikesEnd = _report->getEndTime();
+        // We need _hasEnded to return true  is file based reports because
+        // otherwise getFrameRange doesn't work as expected before the end
+        // of the report is reached. The only way to detect this with the new
+        // API is by checking if the endTime is already different from 0 at
+        // construction, because for streams it will always be 0. This is
+        // relying on an implementation detail of the spikeReport, so it's
+        // worth considering to include this detail as part of the API or add
+        // the isStream function again;
```

Comment:
This indicates still something wrong with the API. From what I can see before both start and end-time were 0 in the streamcase, so I wonder why getFrameRange() behaves differently then.